### PR TITLE
fix: match-all filter in directory app

### PR DESCRIPTION
**Pull request by konstantina-gkp**
fixes: #3393
Adjusted emx2 query builder to support “match-all” filter.
Initially the query for “match-all” filter was:

```{
Biobanks(orderby: { name: ASC }, filter: { _and: [ { collections: { categories: { name: { equals: ["covid19", "paediatric_only"] } } } } ] }) {
    juridical_person,
    id,
    name,
    withdrawn,
    quality {,
        quality_standard {
            name
        }
    },
    collections(orderby: { id: ASC }, filter: { _and: [ { categories: { name: { equals: ["covid19", "paediatric_only"] } } } ] }) {
        id,
        size,
        name,
        withdrawn,
        type {
            label,
            name
```
Now the query for "match-all"is:
```
{
Biobanks(orderby: { name: ASC }, filter: { _and: [ { collections: { categories: { name: { equals: "covid19" } } } }, { collections: { categories: { name: { equals: "paediatric_only" } } } } ] }) {
    juridical_person,
    id,
    name,
    withdrawn,
    quality {,
        quality_standard {
            name
        }
    },
    collections(orderby: { id: ASC }, filter: { _and: [ { categories: { name: { equals: "covid19" } } }, { categories: { name: { equals: "paediatric_only" } } } ] }) {
        id,
        size,
        name,
        withdrawn,
        type {
            label,
            name
```

### CODE REVIEW

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
I've tested several filter combinations, some of them go OK, some of then don't

- Country => when I select two country to match all => result should be no biobanks found, but query gets stuck (infinite). This also holds for other filters where I select more options and the result should be no biobanks found.
- When I select Categories to match all paediatric and covid19, result = 0 collections, count is indeed 0, but 5 biobanks are shown (this should be none). This also happens with other options.
- ~~When I select paediatric_included and covid19, counts are OK, but when I than select a third one (rare_disease) there should be no biobanks found, however, 1 biobank with one (sub) collection is shown.~~ This is actual a bug in the production version!

**Comment by konstantina-gkp** [comment]
This is a separate issue and not related to the current fix/issue. The "match-all" option works as intended. I have reported the new issue [here](https://github.com/molgenis/molgenis-emx2/issues/3445).

**Review by dtroelofsprins — APPROVED** [review]

File: apps/molgenis-components/src/queryEmx2/queryEmx2.ts [hunk 1]
```diff
@@ -474,33 +474,28 @@ class QueryEMX2 {
   _createFilterFromPath(path: string, operator: string, value: any) {
     const valueArray = Array.isArray(value) ? value : [value];
 
-    for (const value of valueArray) {
-      /** reverse the path, so we can build it from the inside out */
-      const reversedPathParts = path.split(".").reverse();
-      let graphqlValue = typeof value === "boolean" ? `${value}` : `"${value}"`;
+    /** set the correct operators for graphQl */
+    if (operator === "in") {
+      operator = "equals";
```
Comment (chinook25) [inline apps/molgenis-components/src/queryEmx2/queryEmx2.ts:479]: Don't do input modification.
I suggest making a getGqlOperator function for this

File: apps/molgenis-components/src/queryEmx2/queryEmx2.ts [hunk 2]
```diff
@@ -474,33 +474,28 @@ class QueryEMX2 {
...
-      }
+    if (operator === "orLike") {
+      operator = "like";
+    }
 
-      if (operator === "in") {
-        graphqlValue = `["${valueArray.join('", "')}"]`;
-        operator = "equals";
-      }
+    for (const value of valueArray) {
```
Comment (chinook25) [inline apps/molgenis-components/src/queryEmx2/queryEmx2.ts:486]: You're shadowing `value` here, don't do that unless there is a good reason to.

File: apps/molgenis-components/src/queryEmx2/queryEmx2.ts [hunk 3]
```diff
@@ -474,33 +474,28 @@ class QueryEMX2 {
...
+      let graphqlValue = typeof value === "boolean" ? `${value}` : `"${value}"`;
 
-      /** most inner part of the query e.g. 'like: "red" */
       let filter = `{ ${operator}: ${graphqlValue} }`;
 
+      /** reverse the path, so we can build it from the inside out */
```
Comment (chinook25) [inline apps/molgenis-components/src/queryEmx2/queryEmx2.ts:491]: try to avoid terms like `it` in comments. I guess it's `the filter`?

**Review by chinook25 — CHANGES_REQUESTED** [review]
some code technical comments, but functionally it works :)

File: apps/molgenis-components/src/queryEmx2/queryEmx2.ts [hunk 4]
```diff
@@ -471,36 +471,36 @@ class QueryEMX2 {
...
       }
-      this.filters[this.branch][queryType].push(filter);
 
-      /** we folded all into one so just return */
-      if (graphqlValue.includes("[")) return;
+      this.filters[this.branch][this.type].push(filter);
+    }
+  }
+
+  /** set the correct operators for graphQl */
```
Comment (chinook25) [inline apps/molgenis-components/src/queryEmx2/queryEmx2.ts:495]: This comment does not correctly describe what the function does, it gets, not sets. Please remove, the function name already sufficiently tells what it does. 
In general, only use comments to explain weird code that can't be written comprehensively (e.g. regexes) or for references.

**Review by chinook25 — APPROVED** [review]
Just one small thing, if you remove the comment you can merge :)

**Pull request merged**